Tests/SMSCounterTest.php

```diff
@@ -182,17 +219,17 @@ public function testUnicode()
 
     public function testUnicodeEmoji()
     {
-        $text = '😎😎';
+        $text = '😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎😎';
 
         $smsCounter = new SMSCounter();
         $count = $smsCounter->count($text);
 
         $expected = new \stdClass();
         $expected->encoding = SMSCounter::UTF16;
-        $expected->length = 2;
-        $expected->per_message = 70;
-        $expected->remaining = 68;
-        $expected->messages = 1;
+        $expected->length = 132;
+        $expected->per_message = 67;
+        $expected->remaining = 2;
```

Comment:
Not sure if we want to do anything about this, but there's a funny situation here because the emoji are twice as long. We say that two are remaining, but in actuality you can only append 1 more (regular) character and stay within 2 messages. If you type 2 more (or 1 more emoji) then it gets moved to a third segment. The reason for this is that the per-message limit is an odd number, so there's actually 1 character's worth of space unused at the end of the first segment. So the "2 remaining" is actually 1 in the first segment and 1 in the second.

Reply:
Well, this is indeed an odd situation

Currently remaining char count are checked against text length as a whole, an improvement could be checking length chunk by chunk and return one of this two options:

* Return only what is left in last chunk, even though it will in the end be strange anyway as _length_ + _remaining_ will not match parts total length `132+1 !== 67*2`
* Return a new value _remaining_last_part_ returning that one single char at the end and leaving _remaining_ as it is

Any other idea?

Reply:
If I had to guess about most common use cases and the most intuitive fix, I'd say the first option is better even though the sum will be off. The behavior should be documented, though.